### PR TITLE
Update CSharpSyntaxFacts for records

### PR DESCRIPTION
**Pull request by Youssef1313**
Not sure how/where to test these.

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs [hunk 1]
```diff
@@ -1878,6 +1881,8 @@ public override DeclarationKind GetDeclarationKind(SyntaxNode declaration)
             {
                 case SyntaxKind.ClassDeclaration:
                     return DeclarationKind.Class;
+                case SyntaxKind.RecordDeclaration:
+                    return DeclarationKind.Record;
```
Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs:1885]: This has an error: `'DeclarationKind' does not contain a definition for 'Record'`

Searching through the code, there are:

- A `DeclarationKind` enum in `Microsoft.CodeAnalysis.CSharp` namespace, which contain `Record` member.
- A `DeclarationKind` enum in `Microsoft.CodeAnalysis.Editing` namespace, which doesn't contain `Record` member.

I'm not sure what to do here?

Should I just return DeclarationKind.Class for both classes and records?

Note that this public method:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.editing.syntaxgenerator.getdeclarationkind?view=roslyn-dotnet

is dependent on this code path.

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs:1885]: > A DeclarationKind enum in Microsoft.CodeAnalysis.Editing namespace, which doesn't contain Record member.

Fix this up.

Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxFacts.cs:1885]: This being fixed in #48096

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Is this ready for merge?

**Pull request merged**